aws/resource_aws_ssm_parameter_test.go

```diff
@@ -273,26 +331,88 @@ resource "aws_ssm_parameter" "foo" {
 func testAccAWSSSMParameterBasicConfigOverwrite(rName, pType, value string) string {
 	return fmt.Sprintf(`
 resource "aws_ssm_parameter" "foo" {
-  name  = "%s"
-  type  = "%s"
-  value = "%s"
+  name  = "test_parameter-%[1]s"
+  description  = "description for parameter %[1]s"
+  type  = "%[2]s"
+  value = "%[3]s"
   overwrite = true
 }
 `, rName, pType, value)
 }
 
-func testAccAWSSSMParameterSecureConfigWithKey(rName string, value string) string {
+func testAccAWSSSMParameterBasicConfigOverwriteWithoutDescription(rName, pType, value string) string {
+	return fmt.Sprintf(`
+resource "aws_ssm_parameter" "foo" {
+  name  = "test_parameter-%[1]s"
+  type  = "%[2]s"
+  value = "%[3]s"
+  overwrite = true
+}
+`, rName, pType, value)
+}
+
+func testAccAWSSSMParameterSecureConfig(rName string, value string) string {
 	return fmt.Sprintf(`
 resource "aws_ssm_parameter" "secret_foo" {
-  name  = "test_secure_parameter-%s"
+  name  = "test_secure_parameter-%[1]s"
+  description  = "description for parameter %[1]s"
   type  = "SecureString"
-  value = "%s"
-	key_id = "${aws_kms_key.test_key.id}"
+  value = "%[2]s"
+}
+`, rName, value)
+}
+
+func testAccAWSSSMParameterSecureConfigWithKey(rName string, value string, keyAlias string) string {
+	return fmt.Sprintf(`
+resource "aws_ssm_parameter" "secret_foo" {
+  name  = "test_secure_parameter-%[1]s"
+  description  = "description for parameter %[1]s"
+  type  = "SecureString"
+  value = "%[2]s"
+  key_id = "alias/%[3]s"
```

Comment:
Minor nitpick: `${aws_kms_alias.test_alias.name}` is a little easier to understand here instead of depends_on 😉